### PR TITLE
adding NodeModulesState type

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**

### CODE REVIEW

File: api/v1beta1/nodemodulestate_types.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ModuleNodeSpec struct {
+	Name            string                   `json:"name"`
+	Namespace       string                   `json:"namespace"`
+	ModuleLoader    ModuleLoaderSpec         `json:"moduleLoader"`
+	ImageRepoSecret *v1.LocalObjectReference `json:"imageRepoSecret,omitempty"`
+	KmomLoaded      bool                     `json:"kmodLoaded"`
```
Comment (qbarrand) [inline api/v1beta1/nodemodulestate_types.go:29]:
```suggestion
	KmodLoaded      bool                     `json:"kmodLoaded"`
```

File: api/v1beta1/nodemodulestate_types.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+// NodeModulesStateSpec describes the desired state of modules on the node
+// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+// +kubebuilder:validation:Required
+type NodeModulesStateSpec struct {
+	// ModuleStateSpec list the spec of all the modules that need to be executed
+	// on the node
+	Modules []ModuleNodeSpec `json:"modules,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+type ModuleNodeStatus struct {
```
Comment (qbarrand) [inline api/v1beta1/nodemodulestate_types.go:41]: I think we need a `ModuleLoaderSpec` as well here, to determine if e.g. the desired parameters are different from those the module was loaded with.

File: api/v1beta1/nodemodulestate_types.go (hunk 1 above)
Comment (qbarrand) [reply api/v1beta1/nodemodulestate_types.go:29]: Do we need a full `ModuleLoaderSpec`? In my PoC I had
```go
type ModuleConfig struct {
	ContainerImage string `json:"containerImage"`

	//+optional
	InTreeModuleToRemove string `json:"inTreeModuleToRemove"`

	Modprobe ModprobeSpec `json:"modprobe"`
}
```

Comment (yevgeny-shnaidman) [reply api/v1beta1/nodemodulestate_types.go:29]: ok, i can change it, we will update later if needed

File: api/v1beta1/nodemodulestate_types.go (hunk 2 above)
Comment (yevgeny-shnaidman) [reply api/v1beta1/nodemodulestate_types.go:41]: I am not sure if we want to support such a flow. Changing the parameters requires module unloading and reloading. It seems more like an upgrade flow

File: api/v1beta1/nodemodulestate_types.go (hunk 1 above)
Comment (yevgeny-shnaidman) [reply api/v1beta1/nodemodulestate_types.go]: fixed

**Pull request merged**